x-pack/plugins/cases/public/components/system_actions/hooks/use_alert_data_view.ts

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import type { DataView } from '@kbn/data-views-plugin/common';
+import { useKibana } from '@kbn/kibana-react-plugin/public';
+import type { ValidFeatureId } from '@kbn/rule-data-utils';
+import { AlertConsumers } from '@kbn/rule-data-utils';
+import { useEffect, useMemo, useState } from 'react';
+import { useQuery } from '@tanstack/react-query';
+import type { TriggersAndActionsUiServices } from '@kbn/triggers-actions-ui-plugin/public';
+import { fetchAlertIndexNames } from './alert_index';
+import { fetchAlertFields } from './alert_fields';
+
+export interface UserAlertDataViews {
+  dataViews?: DataView[];
+  loading: boolean;
+}
+
+export function useAlertDataViews(featureIds: ValidFeatureId[]): UserAlertDataViews {
```

Comment:
This is a hook from `x-pack/plugins/triggers_actions_ui/public/application/hooks/use_alert_data_view.ts`, we should use package for this hook once it is ready.

Reply:
I know it's a temporary solution, but if you want to avoid copying this code (and since `cases` already has a dependency on `triggersAcitonsUI`) I guess it wouldn't hurt to export the hook from the plugin so you can use it directly, what do you say?

Reply:
I can export it as long as it's okay for RAM team 🙂

Reply:
@umbopepato Do you mean to export it from the `start` method of the `plugin.ts` or just export it from the `common` folder so we can import it in `cases`?

Reply:
It can be used in isolation so the second option would work as well. Don't know if @XavierM would be ok with this but it seems a good temporary tradeoff to me